### ISSUE TITLE
Random failures in nightly wheel upload action

### ISSUE THREAD

**Issue by MridulS** [networkx/networkx#7215]
Not sure what is causing this but the action https://github.com/networkx/networkx/actions/workflows/nightly.yml failed a couple of times during this week. Manually re-triggering makes the action work just fine.

From the error, seems like the action was uploading and deleting it at the same time?
```
Uploading file "scientific-python-nightly-wheels/networkx/3.3rc0.dev0/networkx-3.3rc0.dev0-py3-none-any.whl"
Warning:  Distribution "networkx-3.3rc0.dev0-py3-none-any.whl" already exists. Removing.
Error:  ("release version='3.3rc0.dev0' does not exist", 404)
```

I am not sure if this is just a connectivity bleep?

**Comment by rossbar** [comment]
Looks like these are still failing - or at least every other run. Very strange.

**Comment by rossbar** [comment]
This will hopefully be addressed by #7309 - let's keep an eye on it for a couple days and see if the red x's go away!

**Comment by rossbar** [comment]
I believe the specific issues that were causing this have been addressed by #7311 . If new uploader issues crop up let's open a fresh issue!
